go/vt/vttablet/tabletmanager/vreplication/vcopier_atomic.go

```diff
@@ -205,7 +200,12 @@ func (vc *vcopier) copyAll(ctx context.Context, settings binlogplayer.VRSettings
 		log.Infof("copying table %s with lastpk %v", tableName, lastpkbv)
 		// Prepare a vcopierCopyTask for the current batch of work.
 		currCh := make(chan *vcopierCopyTaskResult, 1)
-		currT := newVCopierCopyTask(newVCopierCopyTaskArgs(resp.Rows, resp.Lastpk))
+
+		resp2 := resp
+		if parallelism > 1 {
+			resp2 = resp.CloneVT()
+		}
+		currT := newVCopierCopyTask(newVCopierCopyTaskArgs(resp2.Rows, resp2.Lastpk))
```

Comment:
Now that we know this was the bug, we can simplify the code:
```suggestion
		if parallelism > 1 {
			resp = resp.CloneVT()
		}
		currT := newVCopierCopyTask(newVCopierCopyTaskArgs(resp.Rows, resp.Lastpk))
```